lib/services/parser.go

```diff
@@ -518,3 +522,129 @@ func newParserForIdentifierSubcondition(ctx RuleContext, identifier string) (pre
 		},
 	})
 }
+
+// NewResourceParser returns a parser made for boolean expressions based on a
+// json-serialiable resource. Customized to allow short identifiers common in all
+// resources:
+//  - `metadata.name` can be referenced with `name` ie: `name == "jenkins"``
+//  - `metadata.labels + spec.dynamic_labels` can be referenced with `labels`
+//     ie: `labels.env == "prod"`
+// All other fields can be referenced by starting expression with identifier `resource`
+// followed by the names of the json fields ie: `resource.spec.public_addr`.
+func NewResourceParser(resource types.ResourceWithLabels) (BoolPredicateParser, error) {
+	p, err := predicate.NewParser(predicate.Def{
+		Operators: predicate.Operators{
+			AND: predicate.And,
+			OR:  predicate.Or,
+			NOT: predicate.Not,
+			EQ: func(a interface{}, b interface{}) predicate.BoolPredicate {
+				return func() bool {
+					switch aval := a.(type) {
+					case label:
+						bval, ok := b.(string)
+						return ok && aval.value == bval
+					case string:
+						bval, ok := b.(string)
+						return ok && aval == bval
+					default:
+						return false
+					}
+				}
+			},
+			NEQ: func(a interface{}, b interface{}) predicate.BoolPredicate {
+				return func() bool {
+					switch aval := a.(type) {
+					case label:
+						bval, ok := b.(string)
+						return ok && aval.value != bval
+					case string:
+						bval, ok := b.(string)
+						return ok && aval != bval
+					default:
+						return false
```

Comment:
Should this be `true` instead? Or perhaps the whole `NEQ` could be made into a negation of the function in `EQ`?